yt_dlp/extractor/abc.py

```diff
@@ -299,47 +329,60 @@ def tokenize_url(url, token):
                 'hdnea': token,
             })
 
-        for sd in ('1080', '720', 'sd', 'sd-low'):
-            sd_url = try_get(
-                stream, lambda x: x['streams']['hls'][sd], str)
-            if not sd_url:
-                continue
-            formats = self._extract_m3u8_formats(
-                tokenize_url(sd_url, token), video_id, 'mp4',
-                entry_protocol='m3u8_native', m3u8_id='hls', fatal=False)
-            if formats:
-                break
+        formats = []
+        # hls: pre-merged formats
+        # hls-latest: same as hls, but sometimes has separate audio tracks
+        # Note: pre-merged formats in hls-latest are treated as video-only if audio-only tracks
+        # are present, so we extract both types
+        for label in ('hls', 'hls-latest'):
+            for sd_url in traverse_obj(stream, ('streams', label, ('1080', '720', 'sd', 'sd-low'), {str})):
+                fmts = self._extract_m3u8_formats(
+                    tokenize_url(sd_url, token), video_id, 'mp4',
+                    entry_protocol='m3u8_native', m3u8_id=label, fatal=False)
+                if fmts:
+                    formats.extend(fmts)
+                    break
```

Comment:
is there a reason why we need to hardcode all the quality keys? could we just do this?

```suggestion
            for sd_url in traverse_obj(stream, ('streams', label, ..., {url_or_none})):
                fmts = self._extract_m3u8_formats(
                    tokenize_url(sd_url, token), video_id, 'mp4', m3u8_id=label, fatal=False)
                if fmts:
                    formats.extend(fmts)
                    break
```

if there is a good reason for hardcoding them, then let's still transform with `url_or_none` and remove the default `entry_protocol` param:

```suggestion
            for sd_url in traverse_obj(stream, ('streams', label, ('1080', '720', 'sd', 'sd-low'), {url_or_none})):
                fmts = self._extract_m3u8_formats(
                    tokenize_url(sd_url, token), video_id, 'mp4', m3u8_id=label, fatal=False)
                if fmts:
                    formats.extend(fmts)
                    break
```

&nbsp;

Reply:
also wondering about this? are there stream keys we want to avoid?